### PR TITLE
feat: archive datamanager

### PR DESCRIPTION
**Pull request by cboillet-dev**
## Description

New sdk endpoint to archive and dearchive a datamanager see [asana card](https://app.asana.com/0/1201599852028640/1202944294601335/f)

When a compute task try to use an archived data_manager in raises an invalid context error.

## How has this been tested?

- unit tests

## Checklist

- [x ] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

## Connected-PR

- substra-backend [#521](https://github.com/Substra/substra-backend/pull/521) -> this PR
- orchestrator [#54](https://github.com/Substra/orchestrator/pull/54)
- substra sdk [#35](https://github.com/Substra/substra/pull/305)

### CODE REVIEW

**Comment by cboillet-dev** [comment]
/e2e --refs orchestrator=feat/archive-datamanager,substra=feat/archive-datamanager

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by cboillet-dev** [comment]
/e2e --refs orchestrator=feat/archive-datamanager,substra=feat/archive_datamanager

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: backend/orchestrator/mock.py [hunk 1]
```diff
@@ -27,6 +30,17 @@ class Meta:
     parent_task_output_identifier = None
 
 
+class ComputeTaskInputAssetFactory(factory.Factory):
```
Comment (cboillet-dev) [inline backend/orchestrator/mock.py:33]: the factory is used to test the context

File: backend/api/views/filters_utils.py [hunk 2]
```diff
@@ -141,3 +143,15 @@ def filter_queryset(self, request, queryset, view):
...
         return queryset.filter(**django_filters)
+
+
+class ArchiveFilter(BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        # apply filters to queryset
+        if view.action == "list":
+            archived = to_bool(request.query_params.get("archived"))
+            if archived:
+                return queryset
```
Comment (sergebouchut2) [inline backend/api/views/filters_utils.py:154]: Why we are not filtering on `archived=True` here?

Comment (sergebouchut2) [reply backend/api/views/filters_utils.py:154]:
```python
archived = request.query_params.get("archived")
if view.action == "list" and archived is not None:
    queryset = queryset.filter(archived=to_bool(archived))
return queryset
```

Comment (sergebouchut2) [reply backend/api/views/filters_utils.py:154]: Note that I am quite sure we don't need to use custom code and rely on a BooleanField.

Comment (sergebouchut2) [reply backend/api/views/filters_utils.py:154]: Adding it to the datamanager filterset
```python
class DataManagerFilter(FilterSet):
    ...
    archived = BooleanFilter(field_name="archived", ...)
```

Comment (cboillet-dev) [reply backend/api/views/filters_utils.py]: Good catch it simplify the filtering, I also added a test on listing archived asset and the archived field as part of the serializer in [`f879ca`](https://github.com/Substra/substra-backend/pull/521/commits/f879ca257ef6b577c2ecb7f052a18df17aa1a985)

**Comment by guilhem-barthes** [comment]
Thank you for your contribution! Unfortunately, due to organisational and roadmap changes, this item has been deprioritised. We currently have no bandwidth to cope with the code drift between the current code state in the last release and this PR, therefore we are closing it.

**Pull request closed without merging**